src/main/kotlin/org/rust/lang/core/types/infer/TypeInference.kt

```diff
@@ -970,10 +971,20 @@ class RsFnInferenceContext(
     }
 
     private fun inferStructTypeArguments(literal: RsStructLiteral, typeParameters: Substitution) {
-        literal.structLiteralBody.structLiteralFieldList.mapNotNull { field ->
-            field.expr?.let { expr ->
-                val fieldType = field.type
-                expr.inferTypeCoercableTo(fieldType.substitute(typeParameters))
+        literal.structLiteralBody.structLiteralFieldList.filterNotNull().forEach { field ->
+            val fieldType = field.type.substitute(typeParameters)
+            val expr = field.expr
+
+            if (expr != null) {
+                expr.inferTypeCoercableTo(fieldType)
+            } else {
+                // Handle struct field shorthand by looking up the matching declaration in scope.
+                RsCodeFragmentFactory(field.project).createPath(field.referenceName, field)?.let { path ->
+                    resolvePath(path, lookup).forEach { bound ->
```

Comment:
A path is *resolved* only if list of resolved element contains single element. 
```kotlin
val local = resolvePath(path, lookup).singleOrNull()?.element
```

Reply:
Okay, I'll change that. If it contains multiple elements, what does that mean?

Reply:
It means that we resolved to multiple declarations, that in general means that the path is NOT resolved and our (rust) code is invalid. E.g.
```rust
struct S;
struct S;
type T = S; // multi-resolve here. 
```
In the case of local variables the case of multi-resolve is mostly impossible (b/c of name shadowing), but let's follow general style and treat multi-resolve as it were not resolved